lib/article_status_manager.rb

```diff
@@ -12,25 +12,20 @@ def initialize(wiki = nil)
   # Entry points #
   ################
 
-  # Queries deleted state and namespace for all articles from current courses.
-  def self.update_article_status
-    threads = Course.current
-                    .in_groups(Replica::CONCURRENCY_LIMIT, false)
-                    .map.with_index do |course_batch, i|
-      Thread.new(i) do
-        course_batch.each do |course|
-          update_article_status_for_course(course)
-        end
-      end
-    end
-    threads.each(&:join)
-  end
-
   def self.update_article_status_for_course(course)
     Wiki.all.each do |wiki|
```

Comment:
It looks like this strategy of going through every wiki for every course is left over from before Amit implemented `CoursesWikis`. This should be `course.wikis.each` instead; we only ever count contributions to tracked wikis for each course, so there's no need to deal with pages edited on other wikis.

Reply:
Done!